JoinColony/colonyCDapp
# [Multi-Sig] fix: user popovers not displaying inherited permissions

## Description

What it says, since inherited roles aren't in the database, we need to do some workarounds on the UI for them.
Paired this w/ @mmioana :hand: 

## Testing

I think the following cases should cover everything.

Preparations: 

1. Give `amy` the architecture role in `General`
![image](https://github.com/user-attachments/assets/74842c00-c551-4840-a5f7-36616057cbe2)
2. Give `amy` the funding role in `Andromeda`
![image](https://github.com/user-attachments/assets/8005e861-21f5-488c-9a77-1fceb9cd966c)
3. Give `fry` the `Payer` role set in `General`
![image](https://github.com/user-attachments/assets/05e4a8aa-7c84-4f7d-a307-54874d3c04ee)
4. Give `fry` the `Admin` role set in `Andromeda`
5. Give `leela` the `Architecture` role in `Andromeda`
![image](https://github.com/user-attachments/assets/084493d2-9999-4004-b9d3-3822a86a5980)
6. Install the Multi-Sig extension
7. Give `leela` the `Owner` Multi-Sig role
![image](https://github.com/user-attachments/assets/5182142e-3e99-4195-9860-1a20df6f3a4e)
8. Give `amy` `Funding` Multi-Sig role in `Andromeda`
![image](https://github.com/user-attachments/assets/9bcf204d-20f6-4db3-8f54-76db35eba135)
9. Give `amy` the `Architecture` Multi-Sig role in `General`
![image](https://github.com/user-attachments/assets/36121428-9a59-4a74-843f-3b9f9b7b3437)

Testing:
1. Verify that `leela` has `Owner` bundle in general and `Owner` in Multi-Sig
![image](https://github.com/user-attachments/assets/f8cdb6c2-b093-4540-a3a8-85a2d6300523)
![image](https://github.com/user-attachments/assets/cd812e08-51aa-4ca8-af18-7acd96e4b6fd)
2. Verify that `leela` has `Architecture` in `Andromeda` but inherits all owner roles except recovery and root
![image](https://github.com/user-attachments/assets/31fbead2-1448-4cfa-ad14-11630ab31586)
3. Verify that `amy` has `Architecture` in `General` and Multi-Sig `Architecture` in `General`
![image](https://github.com/user-attachments/assets/bb3f3866-4154-4341-bc1b-ae428b33ceb2)
![image](https://github.com/user-attachments/assets/12a1a945-2e08-48c9-b9e3-ade7d5d2f6f9)
4. Verify that `amy` has `Funding` and `Architecture (inherited)` in `Andromeda` (same for Multi-Sig)
![image](https://github.com/user-attachments/assets/09e2bd14-b542-4b14-8686-49a8b29ef2c6)
![image](https://github.com/user-attachments/assets/929d9f4f-42f8-4293-aff9-fc6aea8cbc66)
5. Verify that `amy` has just `Architecture (inherited)` in any other subdomain (same for Multi-Sig)
![image](https://github.com/user-attachments/assets/e26d6258-8749-4b08-8988-aedc9abf9e5f)
![image](https://github.com/user-attachments/assets/432898bd-81e6-44a5-94e0-a22bdb5a8434)
6. Verify that `fry` has the `Payer` bundle in `General`
![image](https://github.com/user-attachments/assets/6cf98343-a8c6-46ea-bf9b-aadaefa2c937)
7. Verify that `fry` has the `Admin` bundle in `Andromeda`
![image](https://github.com/user-attachments/assets/1aaa8df3-3f00-4b3c-86c7-1a5edc485546)
8. Verify that `fry` has 3 inherited permissions in any other subdomain
![image](https://github.com/user-attachments/assets/a638b305-d8c6-4bea-a288-4cd591f6defe)

## Diffs

**Changes** 🏗

* Decoupled the `DomainPermissionsList` component
* Decoupled the `PermissionTooltip` component
* And also decoupled the `PermissionTooltipContent` component :D 
* If a user has only non inherited roles, display `This member has the {role} permission bundle that includes the following permissions:`, otherwise display `This member has the following permission types:`
* Display the `(inherited)` appendix for inherited roles

Resolves  #2783
